android-components/components/browser/state/src/main/java/mozilla/components/browser/state/engine/middleware/TranslationsMiddleware.kt

```diff
@@ -48,6 +48,7 @@ class TranslationsMiddleware(
             is InitAction ->
                 scope.launch {
                     requestEngineSupport(context)
+                    requestSupportedLanguages(context)
```

Comment:
If the initial `requestEngineSupport()` fails, should we even bother calling `requestSupportedLanguages()`?

Reply:
Thanks, really good point! So, updated to make `requestEngineSupport` take a job of the remaining things to `init` `onSuccess`. (I have a feeling this area will need further refinement, but I think this should be good for now. No change in testing and I manually verified against Fenix.)

The fallback scenario is something I'm still thinking about. It is pretty clear what to do if we know we support the engine, more unclear if it is unknown.